app/src/main/java/de/ph1b/audiobook/features/settings/dialogs/PlaybackSpeedDialogController.kt

```diff
@@ -53,25 +57,31 @@ class PlaybackSpeedDialogFragment : DialogFragment() {
     seekBar.progress = ((speed - MIN) * FACTOR).toInt()
 
     // observable of seek bar, mapped to speed
-    @Suppress("CheckResult")
-    seekBar.progressChangedStream(initialNotification = true)
-      .map { Book.SPEED_MIN + it.toFloat() / FACTOR }
-      .doOnNext {
-        // update speed text
-        val text = "${getString(R.string.playback_speed)}: ${speedFormatter.format(it)}"
-        textView.text = text
-      }
-      .debounce(50, TimeUnit.MILLISECONDS, AndroidSchedulers.mainThread())
-      .subscribe { playerController.setSpeed(it) } // update speed after debounce
+    compositeDisposable.add(
+      seekBar.progressChangedStream(initialNotification = true)
+        .map { Book.SPEED_MIN + it.toFloat() / FACTOR }
+        .doOnNext {
+          // update speed text
+          val text =
+            "${activity!!.getString(R.string.playback_speed)}: ${speedFormatter.format(it)}"
+          textView.text = text
+        }
+        .debounce(50, TimeUnit.MILLISECONDS, AndroidSchedulers.mainThread())
+        .subscribe { playerController.setSpeed(it) }) // update speed after debounce
```

Comment:
It is usually bad practice to subscribe without clearing the `Disposable` at some point. I thought I would add this here.

Reply:
Let's move that logic to the `DialogController` and add an extension function `Disposable.disposeOnDestroyDialog`.

The implementation detail that's making this buggy is that `onCreateDialog` is called from the controllers `onCreateView`. But you dispose now in `onDetach`. 
When we are now in a situatio occurs where `onDetach` is called, but no `onDestroyView`, then we'll dispose the progress stream, but don't observe it when `onAttach` is called.

You can test this by opening the dialog, click on the home button and open the app again. At this point, the speed won't change upon sliding.

In general the subscription only needs to be disposed if it has a reference to the activity for example. Else it will be garbage collected along with the view. It's basically the same like with an `OnClickListener`. You don't remove all `OnClickListener`s when a view gets destroyed too.

Reply:
That makes sense. I'm not familiar with the `conductor` lifecycle. Thank you!